hal/ia32/pmap.c

```diff
@@ -235,40 +226,48 @@ int pmap_getPage(page_t *page, addr_t *addr)
 	page->addr = a;
 	page->flags = 0;
 
-	for (k = 0; k < syspage->mmsize; k++) {
+	map = syspage_mapList();
+	if (map == NULL)
+		return -EINVAL;
 
-		if ((a >= syspage->mm[k].addr) && (a <= syspage->mm[k].addr + syspage->mm[k].len - 1) &&
-			(a + SIZE_PAGE - 1 >= syspage->mm[k].addr) && (a + SIZE_PAGE - 1 <= syspage->mm[k].addr + syspage->mm[k].len - 1)) {
-			if (syspage->mm[k].attr == 1)
-				page->flags |= PAGE_FREE;
-			break;
+	do {
+		if (a >= map->start && a < map->end) {
+			sysEntry = map->entries;
+			if (sysEntry != NULL) {
+				do {
+					if (!(a >= sysEntry->start && a < sysEntry->end))
+						continue;
+
+					/* Memory reserved for boot rom */
+					if ((sysEntry->type == hal_entryReserved)) {
+						*addr = a + SIZE_PAGE;
+						page->flags |= PAGE_OWNER_BOOT;
+						return EOK;
+					}
+
+					/* Skip invalid entries in map */
+					if (sysEntry->type == hal_entryInvalid) {
+						a = (sysEntry->end & ~(SIZE_PAGE - 1)) - SIZE_PAGE;
+						*addr = a + SIZE_PAGE;
+						return -EINVAL;
+					}
+				} while ((sysEntry = sysEntry->next) != map->entries);
+			}
 		}
-
-		/* Test if next memory area is higher than current page */
-		if ((syspage->mm[k].addr > a) && (syspage->mm[k].addr > a + SIZE_PAGE - 1)) {
-			a = (syspage->mm[k].addr & ~0xfff) - SIZE_PAGE;
-			k = syspage->mmsize;
-			break;
+		else {
+			/* Skip empty area between maps */
+			if (a > (map->end - 1) && a < map->next->start) {
+				a = (map->next->start & ~(SIZE_PAGE - 1)) - SIZE_PAGE;
+				*addr = a + SIZE_PAGE;
+				return -EINVAL;
+			}
 		}
+	} while ((map = map->next) != syspage_mapList());
 
-		/* Test if page overlaps with reserved memory area */
-		if ((a <= syspage->mm[k].addr) && (syspage->mm[k].attr != 1)) {
-			page->flags &= ~PAGE_FREE;
-			break;
-		}
-	}
+	page->flags |= PAGE_FREE;
 	*addr = a + SIZE_PAGE;
 
-	/* If page doesn't exist return error */
-	if (k == syspage->mmsize)
-		return -EINVAL;
-
-	/* Page is allocated by BIOS */
-	if (!(page->flags & PAGE_FREE)) {
-		page->flags |= PAGE_OWNER_BOOT;
-		return EOK;
-	}
-
+#if 0
```

Comment:
You added #if 0, so maybe remove comment that comments-out the code inside?

Reply:
What do you think about removing it permanently? 
This piece of code describes `ptable` which has a new approach in `pmap_getPage` function:

https://github.com/phoenix-rtos/phoenix-rtos-kernel/blob/3b85faa3db0cb156f678c30dc1d430cc9f0a8f00/hal/ia32/pmap.c#L295-L298